### PR TITLE
DYN-6130 remove the toast notification

### PR DESCRIPTION
**Pull request by jesusalvino**
### Purpose

Making sure close the toast notification from the Preferences when Dynamo closes, bug https://jira.autodesk.com/browse/DYN-6130

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Reviewers
@QilongTang 
@RobertGlobant20 

### FYIs
@Enzo707

### CODE REVIEW

**Comment by jesusalvino** [comment]
toast-sandbox

**Comment by jesusalvino** [comment]
toast-D4R

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs [hunk 1]
```diff
@@ -1341,6 +1341,7 @@ public PreferencesViewModel(DynamoViewModel dynamoViewModel)
             this.preferenceSettings.PropertyChanged += PreferenceSettings_PropertyChanged;
             this.pythonScriptEditorTextOptions = dynamoViewModel.PythonScriptEditorTextOptions;
             this.dynamoViewModel = dynamoViewModel;
+            this.dynamoViewModel.Model.ShutdownStarted += Model_ShutdownStarted;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:1344]: ensure that any event subscriptions you add are unsubscribed somewhere else, and make sure that place is actually called when it makes sense.

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs [hunk 2]
```diff
@@ -1428,6 +1429,11 @@ public PreferencesViewModel(DynamoViewModel dynamoViewModel)
             InitializeCommands();
         }
 
+        private void Model_ShutdownStarted(DynamoModel model)
+        {
+            dynamoViewModel.MainGuideManager?.CloseRealTimeInfoWindow();
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:1434]: why is this being called from the preferenceViewModel? IE why is the preferences view model in control of this. It seems pretty strange to me that PreferencesViewModel is gluing these together.

**Comment by mjkkirschner** [comment]
please try to add tests.

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:1434]: Maybe it should be at where we close and dispose `dynamoViewModel`

File: src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@
 using System.Windows.Controls.Primitives;
 using Dynamo.Controls;
 using Dynamo.Logging;
+using Dynamo.Utilities;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs:12]: What is this new reference for?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 4]
```diff
@@ -769,6 +769,12 @@ protected DynamoViewModel(StartConfiguration startConfiguration)
             }
 
             FileTrustViewModel = new FileTrustWarningViewModel();
+            model.ShutdownStarted += Model_ShutdownStarted;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:772]: Isn't there already an event about model disposal or shutdown handling? I was hoping DynamoViewMode -> Dispose -> GuideManager-> Dispose-> Make sure Window resources close

Comment (jesusalvino) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:772]: Probably you were looking for something like the WorkspaceViewModel's Dispose implementation in the DynamoViewModel, but there is not exist.

image

image

Comment (jesusalvino) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: it sounds clean, Done.

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 5]
```diff
@@ -3419,6 +3419,7 @@ public bool PerformShutdownSequence(ShutdownParams shutdownParams)
             WatchHandler.RequestSelectGeometry -= BackgroundPreviewViewModel.AddLabelForPath;
             model.ComputeModelDeserialized -= model_ComputeModelDeserialized;
             model.RequestNotification -= model_RequestNotification;
+            MainGuideManager?.CloseRealTimeInfoWindow();
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:3422]: Thanks @jesusalvino Can you move this line to below `BackgroundPreviewViewModel.Dispose();`? The section below is for DynamoModel shutdown

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
Passed at https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/10887/. Please also cherry-pick to master branch.

**Comment by QilongTang** [comment]
The gif for D4R is way too long, can we **not** demonstrate the host application opening as part of the gif? I would love a gif from only when user launches D4R inside Revit to the end of the experience

**Comment by jesusalvino** [comment]
> The gif for D4R is way too long, can we **not** demonstrate the host application opening as part of the gif? I would love a gif from only when user launches D4R inside Revit to the end of the experience

Done

**Comment by jesusalvino** [comment]
> Passed at https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/10887/. Please also cherry-pick to master branch.

CP https://github.com/DynamoDS/Dynamo/pull/14332

**Comment by jesusalvino** [comment]
> please try to add a regression test.

@mjkkirschner here you are https://github.com/DynamoDS/Dynamo/pull/14340

**Pull request merged**